clients/client-python/tests/unittests/test_gvfs_with_local.py

```diff
@@ -0,0 +1,781 @@
+        with self.assertRaises(RuntimeError):
+            fs._convert_actual_path(actual_path, mock_hdfs_context)
+
+        # test actual path start with storage location
+        actual_path = "/fileset/test_f1/actual_path"
+        virtual_path = fs._convert_actual_path(actual_path, mock_hdfs_context)
+        self.assertEqual(
+            "fileset/test_catalog/test_schema/test_f1/actual_path", virtual_path
+        )
+
+        # test convert actual local path
+        audit_dto = AuditDTO(
+            _creator="test",
+            _create_time="2022-01-01T00:00:00Z",
+            _last_modifier="test",
+            _last_modified_time="2024-04-05T10:10:35.218Z",
+        )
+        local_fileset: FilesetDTO = FilesetDTO(
+            _name="test_f1",
+            _comment="",
+            _type=FilesetDTO.Type.MANAGED,
+            _storage_location="file:/tmp/fileset/test_f1",
+            _audit=audit_dto,
+            _properties={},
+        )
+        mock_local_context: FilesetContext = FilesetContext(
+            name_identifier=NameIdentifier.of_fileset(
+                "test_metalake", "test_catalog", "test_schema", "test_f1"
+            ),
+            storage_type=StorageType.FILE,
+            fileset=local_fileset,
+            actual_path=local_fileset.storage_location() + "/actual_path",
+            fs=LocalFileSystem(),
+        )
+
+        fs = gvfs.GravitinoVirtualFileSystem(
+            server_uri="http://localhost:9090", metalake_name="metalake_demo"
+        )
+        # test actual path not start with storage location
+        actual_path = "/not_start_with_storage/ttt"
+        with self.assertRaises(RuntimeError):
+            fs._convert_actual_path(actual_path, mock_local_context)
+
+        # test actual path start with storage location
+        actual_path = "/tmp/fileset/test_f1/actual_path"
+        virtual_path = fs._convert_actual_path(actual_path, mock_local_context)
+        self.assertEqual(
+            "fileset/test_catalog/test_schema/test_f1/actual_path", virtual_path
+        )
+
+    def test_extract_identifier(self, mock_method1, mock_method2):
+        fs = gvfs.GravitinoVirtualFileSystem(
+            server_uri="http://localhost:9090", metalake_name="metalake_demo"
+        )
+        with self.assertRaises(RuntimeError):
+            fs._extract_identifier(path=None)
+
+        invalid_path = "s3://bucket_1/test_catalog/schema/fileset/ttt"
+        with self.assertRaises(RuntimeError):
+            fs._extract_identifier(path=invalid_path)
+
+        valid_path = "fileset/test_catalog/schema/fileset/ttt"
+        identifier: NameIdentifier = fs._extract_identifier(path=valid_path)
+        self.assertEqual("metalake_demo", identifier.namespace().level(0))
+        self.assertEqual("test_catalog", identifier.namespace().level(1))
+        self.assertEqual("schema", identifier.namespace().level(2))
+        self.assertEqual("fileset", identifier.name())
```

Comment:
Currently, there is no docker environment in Python's integration test. I am considering supporting docker environment + hdfs integration test in a separate PR(#3760), otherwise this PR will become very large.